infrastructure/src/main/java/org/corfudb/infrastructure/LogReplicationServer.java

```diff
@@ -94,43 +91,51 @@ public void shutdown() {
 
     /* ************ Server Handlers ************ */
 
-    @ServerHandler(type = CorfuMsgType.LOG_REPLICATION_ENTRY)
-    private void handleLogReplicationEntry(CorfuPayloadMsg<LogReplicationEntry> msg, ChannelHandlerContext ctx, IServerRouter r) {
+    @RequestHandler(type = PayloadCase.LR_ENTRY_REQUEST)
+    private void handleLrEntryRequest(@Nonnull RequestMsg request,
+                                      @Nonnull ChannelHandlerContext ctx,
+                                      @Nonnull IServerRouter router) {
         log.trace("Log Replication Entry received by Server.");
 
-        if (isLeader(msg, r)) {
+        if (isLeader(request, ctx, router)) {
             // Forward the received message to the Sink Manager for apply
-            LogReplicationEntry ack = sinkManager.receive(msg.getPayload());
+            LogReplication.LogReplicationEntryMsg ack =
+                    sinkManager.receive(request.getPayload().getLrEntryRequest());
 
             if (ack != null) {
-                long ts = ack.getMetadata().getMessageMetadataType().equals(MessageType.LOG_ENTRY_REPLICATED) ?
+                long ts = ack.getMetadata().getEntryType().equals(LogReplicationEntryType.LOG_ENTRY_REPLICATED) ?
                         ack.getMetadata().getTimestamp() : ack.getMetadata().getSnapshotTimestamp();
                 log.info("Sending ACK {} on {} to Client ", ack.getMetadata(), ts);
-                r.sendResponse(msg, CorfuMsgType.LOG_REPLICATION_ENTRY.payloadMsg(ack));
+                CorfuMessage.ResponsePayloadMsg payload = getLogEntryResponse(request);
+                CorfuMessage.ResponseMsg response = getResponseMsg(request.getHeader(), payload);
+                router.sendResponse(response, ctx);
             }
         } else {
             log.warn("Dropping log replication entry as this node is not the leader.");
         }
     }
 
-    @ServerHandler(type = CorfuMsgType.LOG_REPLICATION_METADATA_REQUEST)
-    private void handleMetadataRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+    private CorfuMessage.ResponsePayloadMsg getLogEntryResponse(RequestMsg request) {
+        LogReplication.LogReplicationEntryMsg payload = sinkManager.receive(
+                request.getPayload().getLrEntryRequest());
+        return CorfuMessage.ResponsePayloadMsg.newBuilder()
+                .setLrEntryResponse(payload).build();
+    }
+
```

Comment:
I would like to make them consistent. For example, if you take a look at `getMetadataResponse`, you will notice that it relies on `LogReplicationMetadataManager` for generating its response. So we cannot move it there, since the module where `CorfuProtocolLogReplication` is housed does not have a dependency towards `CorfuProtocolLogReplication` which sits in `infrastructure`.

So I moved them all under its own `LogReplicationServer` sercion.

Reply:
This specific one was moved directly to `LogReplicationMetadataManager`.